src/commands/transform.rs

```diff
@@ -0,0 +1,331 @@
+
+            pipeline = pipeline.with_input_strategy_with_single_source(source);
+            Ok(())
+        }
+        InputSpec::FromMany { inputs, .. } => {
+            let mut expanded_paths = Vec::new();
+
+            for pattern in inputs {
+                for entry in glob::glob(pattern)? {
+                    expanded_paths.push(entry?.to_string_lossy().to_string());
+                }
+            }
+
+            expanded_paths.sort();
+            expanded_paths.dedup();
+
+            if expanded_paths.is_empty() {
+                return Err(anyhow!("No input files found"));
+            }
+
+            let mut sources: Vec<Box<dyn DataSource>> = Vec::new();
+            let mut schema: Option<SchemaRef> = None;
+            for input_path in expanded_paths {
+                let detected_input_format = detect_format(&input_path, input_format)?;
+                let source: Box<dyn DataSource> = match detected_input_format {
+                    DataFormat::Arrow => Box::new(ArrowFileDataSource::new(input_path.clone())),
+                    DataFormat::Parquet => Box::new(ParquetDataSource::new(input_path.clone())),
+                };
+                if let Some(ref schema) = schema {
+                    let source_schema = source.schema()?;
+                    if *schema != source_schema {
+                        return Err(anyhow!(
+                            "Schema mismatch for input file {} (does not match other file(s))",
+                            &input_path
+                        ));
+                    }
+                } else {
+                    schema = Some(source.schema()?);
+                }
+                sources.push(source);
+            }
+            pipeline = pipeline.with_input_strategy_with_multiple_sources(sources);
+            Ok(())
+        }
+    };
+
+    setup_result?;
+
+    let output_spec = match input {
+        InputSpec::From { to, .. } => to,
+        InputSpec::FromMany { to, .. } => to,
+    };
+
+    let list_outputs_format = match &output_spec {
+        OutputSpec::To { .. } => None,
+        OutputSpec::ToMany { list_outputs, .. } => Some(*list_outputs),
+    };
+
+    let partition_columns_for_sort = match &output_spec {
+        OutputSpec::ToMany { by, .. } => Some(
+            by.split(',')
+                .map(|s| s.trim().to_string())
+                .collect::<Vec<_>>(),
+        ),
+        _ => None,
+    };
+
+    match output_spec {
+        OutputSpec::To { output } => {
+            let sink_factory = create_sink_factory(
+                output_format,
+                arrow_compression,
+                arrow_format,
+                arrow_record_batch_size,
+                parquet_compression,
+                parquet_row_group_size,
+                parquet_statistics,
+                parquet_writer_version,
+                parquet_no_dictionary,
+                bloom_filter,
+                parquet_sort_spec,
+            )?;
```

Comment:
we need to delay creating a sink until we know what the schema will be, which it hard to know before we get to that part of the pipeline...